### PR TITLE
grpc: add support for custom channel args.

### PR DESCRIPTION
**Pull request by htuch**
This is useful to allow plumbing of args such as grpc.keepalive_time_ms,
e.g. as a fix for #5173.

Risk level: Low
Testing: Unit and integration tests added.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

File: api/envoy/config/core/v3/grpc_service.proto [hunk 1]
```diff
@@ -203,6 +203,20 @@ message GrpcService {
       }
     }
 
+    // Channel arguments.
+    message ChannelArgs {
```
Comment (lizan) [inline api/envoy/config/core/v3/grpc_service.proto:207]: make this map<string, ChannelArgsValue>?

**Comment by htuch** [comment]
AFAICT the clang-tidy here is a false positive in the unit test.

File: test/common/grpc/google_grpc_utils_test.cc [hunk 2]
```diff
@@ -83,6 +86,38 @@ TEST(GoogleGrpcUtilsTest, ByteBufferInstanceRoundTrip) {
   EXPECT_EQ(buffer_instance2->toString(), "test this");
 }
 
+// Validate that we build the grpc::ChannelArguments as expected.
+TEST(GoogleGrpcUtilsTest, ChannelArgsFromConfig) {
+  envoy::config::core::v3::GrpcService config;
+  auto* grpc_config = config.mutable_google_grpc();
```
Comment (lizan) [inline test/common/grpc/google_grpc_utils_test.cc:92]: nit: use parseYaml<GrpcService>?

**Comment by htuch** [comment]
@lizan I think we should disable this check in favor of likely comparable checks in CodeQL (which @asraa) is working on. Less noise is good :)

**Comment by htuch** [comment]
@lizan anything further on this PR or is it good to merge?

**Pull request merged**